packages/react-aria-components/src/Menu.tsx

```diff
@@ -89,34 +89,50 @@ export interface SubmenuTriggerProps {
   delay?: number
 }
 
+const SubmenuTriggerContext = createContext<{parentMenuRef: RefObject<HTMLElement>} | null>(null);
+
 /**
  * A submenu trigger is used to wrap a submenu's trigger item and the submenu itself.
  *
  * @version alpha
  */
-export function SubmenuTrigger(props: SubmenuTriggerProps, ref: ForwardedRef<HTMLDivElement>): JSX.Element | null {
-  return useSSRCollectionNode('submenutrigger', props, ref, props.children, props.children[0]);
-}
-
-function SubmenuTriggerInner(props) {
-  let {item, parentMenuRef} = props;
+export const SubmenuTrigger =  /*#__PURE__*/ createBranchComponent('submenutrigger', (props: SubmenuTriggerProps, ref: ForwardedRef<HTMLDivElement>, item) => {
```

Comment:
SubmenuTrigger changed so that the state/hooks for submenus live at the trigger level and get passed down to the MenuItem via context rather than having two separate MenuItem implementations. This works due to ref merging by useContextProps.